src/rules.rs

```diff
@@ -359,9 +362,26 @@ fn multiline_string(element: &SyntaxElement) -> bool {
         element.as_node().map(|e| has_newline(&e)).unwrap_or(false);
     }
     false
+}*/
+
+fn has_no_brackets(element: &SyntaxElement) -> bool {
+    let parent = match element.parent() {
+        None => return true,
+        Some(it) => it,
+    };
+    parent.children().all(|it| {
+        it.kind() != NODE_ATTR_SET
+            && it.kind() != NODE_PATTERN
+            && it.kind() != NODE_LAMBDA
+            && it.kind() != NODE_APPLY
```

Comment:
Do we really want `NODE_APPLY` here? I'd say that we probably don't?

Reply:
Yeah actually I'm not confident to put `NODE_APPLY`there because we can have `.inside(NODE_PAREN).before(NODE_APPLY).when(...` I'm still trying to figure out the predicates.

Reply:
One thing that I want to avoid is to have the `fetchurl` got expanded
```
{
  testD = [
    (fetchurl {
      url = "bla";
    })
    (fetchurl {
      url = "foo";
    })
  ];

}
```

Reply:
Aha, `fetchurl` example is interesting. As a general rule, when implementing special cases, it's better to be as specific as possible. In this specific case, just `NODE_APPLY` I think would catch too much. Rather, it seems to me that we want "application, where left hand side is a simple identifier, and the right hand-side is curly-braced identifier". 

So, for APPLY I'd imagine something like this:

```rust
        if let Some(apply) = element.as_node().cloned().and_then(rnix::types::Apply::cast) {
            if let Some(value)  = apply.value() {
                match value.kind() {
                    NODE_ATTR_SET => true,
                    _ => false,
                }
            }
        }
        return false;
```